@xen-orchestra/lite/src/components/pool/dashboard/PoolDashboardCpuProvisionning.vue

```diff
@@ -0,0 +1,99 @@
+<template>
+  <UiCard>
+    <UiTitle type="h4" class="title"
+      >{{ $t("cpu-provisioning") }}
+      <!-- TODO: add a tooltip on the icon when available to explain why there is a warning -->
+      <UiIcon v-if="isWarning" :icon="faWarning"
+    /></UiTitle>
+    <div v-if="isReady" class="progress-item" :class="{ warning: isWarning }">
+      <UiProgressBar color="custom" :value="nVCpuInUse" :max-value="maxValue" />
+      <UiUnitProgressBar :max-value="maxValue" />
+      <UiLegendProgressBar>
+        <template #label>{{ $t("vcpus") }}</template>
+        <template #value>{{ value }}%</template>
+      </UiLegendProgressBar>
+      <UiCardFooter>
+        <template #left>
+          <p>{{ $t("vcpus-used") }}</p>
+          <p class="footer-value">{{ nVCpuInUse }}</p></template
+        >
+        <template #right>
+          <p>{{ $t("total-cpus") }}</p>
+          <p class="footer-value">{{ nPCpu }}</p></template
+        >
+      </UiCardFooter>
+    </div>
+    <UiSpinner v-else class="spinner" />
+  </UiCard>
+</template>
+
+<script lang="ts" setup>
+import { computed } from "vue";
+import { storeToRefs } from "pinia";
+import UiCard from "@/components/ui/UiCard.vue";
+import UiCardFooter from "@/components/ui/UiCardFooter.vue";
+import UiIcon from "@/components/ui/UiIcon.vue";
+import UiLegendProgressBar from "@/components/ui/UiLegendProgressBar.vue";
+import UiProgressBar from "@/components/ui/UiProgressBar.vue";
+import UiUnitProgressBar from "@/components/ui/UiUnitProgressBar.vue";
+import UiSpinner from "@/components/ui/UiSpinner.vue";
+import UiTitle from "@/components/ui/UiTitle.vue";
+import { percent } from "@/libs/utils";
+import { usePoolStore } from "@/stores/pool.store";
+import { useVmMetricsStore } from "@/stores/vm-metrics.store";
+import { useVmStore } from "@/stores/vm.store";
+import { faWarning } from "@fortawesome/free-solid-svg-icons";
+
+const { pool, isReady: poolStoreIsReady } = storeToRefs(usePoolStore());
+const { allRecords: vms, isReady: vmStoreIsReady } = storeToRefs(useVmStore());
+const vmMetricsStore = useVmMetricsStore();
+
+const nPCpu = computed(() => +(pool.value?.cpu_info.cpu_count ?? 0));
+const nVCpuInUse = computed(() =>
+  vms.value
+    .filter((vm) => vm.power_state === "Running" || vm.power_state === "Paused")
+    .reduce(
+      (total, vm) => total + vmMetricsStore.getRecord(vm.metrics).VCPUs_number,
+      0
+    )
+);
```

Comment:
Since there might be a lot of VMs, let's try to optimize this as much as possible by running only one loop instead of 2:
```ts
const ACTIVE_STATES = new Set(["Running", "Paused"])
// ...
vms.value.reduce(
  (total, vm) => ACTIVE_STATES.has(vm.power_state)
    ? total + vmMetricsStore.getRecord(vm.metrics).VCPUs_number
    : total,
  0
)
```